align/compiler/match_graph.py

```diff
@@ -371,18 +371,16 @@ def define_SD(G,power,gnd,clk):
     traversed = []
     probable_changes_p=[]
     if power[0] in G.nodes():
-        while power:
+        high=power.copy()
+        while high:
             try:
-                nxt = power[0]
-                power = power[1:]
-                high=get_next_level(G,[nxt])
-                #logger.debug("next,power: %s %s %s %s",nxt,power,high,traversed)
-                for node in high:
+                nxt = high.pop(0)
+                for node in get_next_level(G,[nxt]):
```

Comment:
@kkunal1408 You were using high in both the inner and outer loop. I am iterating directly over the results of get_next_level instead.

This fixes the infinite loop issue but please check to make sure it has the desired effect.

Reply:
You can look at b417966 for the exact change.